content/doc/book/using/best-practices.adoc

```diff
@@ -0,0 +1,143 @@
+Coverage reports can illustrate areas that automated tests are not executing.
+Compiler warning messages can be a first indication of a problem.
+Static analysis tools can report risky code or code that may have security risks.
+Performance test results can idenitfy delays or problem areas.
+
+The plugin:warnings-ng[warnings next generation] plugin provides convenient access to many reports, including:
+
+* compiler warnnings and errors (like gcc, clang, javac, golang, etc.)
+* static analysis warnings and errors (like spotbugs, checkstyle, pmd, lint, cpd, Simian, etc.)
+* code coverage reports
+
+video::tj3xYFA6Q2o[youtube,width=800,height=420]
+
+== Show failures to the right people
+
+Tune the notifications from failing and unstable jobs so that the notifications reach the right people without distracting others.
+Many Jenkins users prefer to only be notified when a failure is likely their fault.
+They correctly assume that if they did not cause the failure, they are not the best person to investigate the failure.
+
+Refine your notification system so that new test failures first notify the most recent committers, since they are the most likely cause of the issue.
+
+.Sending Slack notifications
+video::EDVZli8GdUM[youtube,width=800,height=420]
+
+== Build on agents
+
+Use agents to perform builds instead of running builds on the controller.
+It is much safer to use agents and much more scalable.
+
+More details are available in the link:/doc/book/security/controller-isolation/[controller isolation] documentation.
+
+== Back up regularly
+
+Even the most reliable of systems will have failures.
+Prepare and regularly check the health of your backups.
+
+More details in the link:/doc/book/system-administration/backing-up/[backup documentation].
+
+== Use simple project names
+
+Jenkins uses project names for folders related to the project.
+Many tools cannot handle spaces, dollar signs, or similar characters in file paths.
+Limit yourself to alphanumeric characters [a-zA-Z0-9_-]+ in project names.
+Use the Display Name feature to make them look nice.
+The "Restrict project naming" setting in the system configuration can enforce naming restrictions for all projects.
+
+== Fingerprint your dependencies
+
+When you have interdependent projects, it often becomes hard to keep track of which version of this is used by which version of that.
+Jenkins supports "file fingerprinting" to simplify this, so make the best use of it.
+
+More information is available in the link:/doc/book/using/fingerprints/[fingerprinting page].
+
+== Strive for reproducible builds
+
+Reproducible builds are a set of software development practices that create an independently-verifiable path from source to binary code.
+A continuous integration build is a possible first step towards reproducible builds, but is usually not sufficient
+
+See the link:https://reproducible-builds.org/[reproducible builds site] for more details.
```

Comment:
```suggestion
Refer to the link:https://reproducible-builds.org/[reproducible builds site] for more details.
```
same as previous suggestions to use"Refer" instead of "See"